### PR TITLE
13956: AUDIT REQUEST - EnrollApp Access List

### PR DESCRIPTION
**Pull request by Battula**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [13956](https://devops.dchbx.org/redmine/issues/13956) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @Ckellem |

### CODE REVIEW

File: script/ea_access_list.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      User.offset(offset).limit(batch_size).each do |user|
+        if processed_count % 1000 == 0
+          puts "processed #{processed_count}" unless Rails.env.test?
+        end
+        begin
+          person = user.person
+          if person.nil?
+            csv << ["No person record found", "No person record", user.roles, user.created_at, user.last_sign_in_at]
+          else
+            csv << [user.person.hbx_id, user.person.full_name, user.roles, user.created_at, user.last_sign_in_at]
```
Comment (blackeaglejs) [inline script/ea_access_list.rb]: I think by User ID they may be referring to the user's OIM ID or the user's email. You may want to provide both. It sounds like this audit is looking for information about user accounts, though it wouldn't hurt to also provide the HBX ID in addition to OIM ID/email.

Comment (blackeaglejs) [inline script/ea_access_list.rb]: Second, the ticket asks for account lock/unlock status and account deactivation date. Neither of these are provided in your version of the report. I'm not entirely sure what these are, but you may want to send them back to the BA to get a better sense of it.

Comment (Battula) [inline script/ea_access_list.rb]: There's no clarity on lock/unlock status and account deactivation date.

Comment (Battula) [inline script/ea_access_list.rb]: I will include both OIM ID and the user's email to the report.

**Pull request merged**